test/cli_stages.c

```diff
@@ -17,6 +17,7 @@
 cli_info_t *cli_info = NULL;
 int cli_info_cnt = 0;
 bool test_abort = false;
+bool test_complete = false;
```

Comment:
@rhc54 IIUC, `test_complete` should be `volatile` to prevent it to be cached in the register. SIGCHLD callback will be invoked in the service thread and the poll happens in the main thread.